### ISSUE TITLE
Jointly training of  a CNN and a LSTM

### ISSUE THREAD

**Issue by ghost** [fchollet/keras#3301]
Hi,

I'm trying to create a model, containing stack convolutions and LSTM cells. To do this, I should distribute the CNN layers over time, and then add the LSTM cell. A small version of this could be like this:

`NN_model = Sequential()
CNN_model.add(ZeroPadding2D((1,1),input_shape=(3,100,100)))
CNN_model.add(Convolution2D(32, 3, 3, activation='relu'))
CNN_model.add(MaxPooling2D((2,2), strides=(2,2)))
CNN_model.add(Flatten())
CNN_model.add(Dense(256, activation='relu'))

TD_input = Input(shape=(INPUT_LEN,3,100,100))
TD_LAYERS = []
ly_temp = TimeDistributed(CNN_model.layers[1])(TD_input)
list.insert(TD_LAYERS,0, ly_temp)

for ly in range(1,5):
    ly_temp = TimeDistributed(CNN_model.layers[ly])(TD_LAYERS[ly-1])
    list.insert(TD_LAYERS,ly, ly_temp)

Layer1_LSTM1 = LSTM(128,return_sequences=False)(TD_LAYERS[4])
Layer1_DO1 = Dropout(0.5)(Layer1_LSTM1)
Layer1_Dense1 = Dense(OUTPUT_LEN)(Layer1_DO1)
Layer1_Activation = Activation('softmax', name='output1')(Layer1_Dense1)

TD_model = Model(input = TD_input, output = Layer1_Activation)
`

When I want to train this model, I get this error:

> this shared variable already has an update expression', (convolution2d_1_W, GpuFromHost.0)

Any idea why I see this error?

Thanks,
Sadegh

**Comment by codekansas** [comment]
You should add a new convolution for each layer instead of reusing the same convolution layer 5 times.

**Comment by Higgcz** [comment]
I don't understand why you're creating the Sequential CNN model at first and using its layers to create TD model, is there some special reason? Because otherwise you can create directly the TD model:

```python
TD2_input = Input(shape=(INPUT_LEN,3,100,100))
TD2_layer = TimeDistributed(ZeroPadding2D((1,1),input_shape=(3,100,100)))(TD2_input)
TD2_layer = TimeDistributed(Convolution2D(32, 3, 3, activation='relu'))(TD2_layer)
TD2_layer = TimeDistributed(MaxPooling2D((2,2), strides=(2,2)))(TD2_layer)
TD2_layer = TimeDistributed(Flatten())(TD2_layer)
TD2_layer = TimeDistributed(Dense(256, activation='relu'))(TD2_layer)

TD2_layer = LSTM(128,return_sequences=False)(TD2_layer)
TD2_layer = Dropout(0.5)(TD2_layer)
TD2_layer = Dense(OUTPUT_LEN)(TD2_layer)
TD2_layer = Activation('softmax', name='output1')(TD2_layer)
TD2_model = Model(input = TD2_input, output = TD2_layer)
```

You don't even need to use the Functional approach to build the model.

However I try to build and learn your model and it works, you have just small typo in your code, you're adding same (first) layer twice: 
```python
...
ly_temp = TimeDistributed(CNN_model.layers[1])(TD_input)
...
```
Shoud be, I guess:
```python
...
ly_temp = TimeDistributed(CNN_model.layers[0])(TD_input)
...
```
